cpp/CMakeLists.txt

```diff
@@ -34,7 +34,7 @@ else()
     else()
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
     endif()
-    set(CMAKE_CXX_FLAGS_DEBUG "-ggdb")
+    set(CMAKE_CXX_FLAGS_DEBUG "-g3 -fsanitize=address -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow")
```

Comment:
I see that it was like that even before your change but explicitly setting CMAKE_CXX_FLAGS* is bad because it prevents people from setting their own debug flags at configure time. I think we should use a generator expression in order to append to the flags rather than overriding them.

```cmake
add_compile_options("$<$<CONFIG:DEBUG>:-g3;-fsanitize=address;-fsanitize=undefined -fsanitize=float-divide-by-zero;-fsanitize=float-cast-overflow>")
```